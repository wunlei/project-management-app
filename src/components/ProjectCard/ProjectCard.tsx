import { Card, CardHeader, IconButton, Typography } from '@mui/material';
import { ReactComponent as TrashIcon } from 'assets/icons/trash.svg';
import { ProjectCardProps } from './ProjectCard.types';

export default function ProjectCard({ color, title }: ProjectCardProps) {
  return (
    <Card
      variant="outlined"
      sx={{
        borderWidth: '2px',
        borderStyle: 'solid',
        borderColor: color,
        borderLeftWidth: '7px',
        maxWidth: '300px',
        '&:hover': {
          cursor: 'pointer',
          boxShadow: 2,
        },
      }}
    >
      <CardHeader
        disableTypography
        title={
          <Typography variant="h5" fontWeight="bold">
            {title}
          </Typography>
        }
        action={
          <IconButton color="error">
            <TrashIcon />
          </IconButton>
        }
      ></CardHeader>
    </Card>
  );
}
