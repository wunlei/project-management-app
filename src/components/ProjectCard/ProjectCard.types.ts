export interface ProjectCardProps {
  color: string;
  title: string;
  description: string;
}
