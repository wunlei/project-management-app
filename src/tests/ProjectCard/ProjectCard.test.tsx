import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import ProjectCard from 'components/ProjectCard/ProjectCard';

it('should render with provided data', () => {
  const data = {
    onDelete: () => {},
    onEdit: () => {},
    title: 'My Project',
    description: 'Project description',
    boardId: '1',
  };
  const card = renderer
    .create(
      <BrowserRouter>
        <ProjectCard
          onDelete={data.onDelete}
          onEdit={data.onEdit}
          title={data.title}
          description={data.description}
          boardId={data.boardId}
        />
      </BrowserRouter>
    )
    .toJSON();
  expect(card).toMatchSnapshot();
});
