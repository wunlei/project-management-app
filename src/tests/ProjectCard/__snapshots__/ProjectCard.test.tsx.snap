// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with provided data 1`] = `
<a
  className="css-1r4kv-MuiStack-root"
  href="/1"
  onClick={[Function]}
>
  <div
    className="css-1wh65h9-MuiStack-root"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5 css-5rh7kk-MuiTypography-root"
    >
      My Project
    </h5>
    <div
      className="css-w4z10b-MuiStack-root"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-lg24u0-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg>
          edit.svg
        </svg>
      </button>
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorError MuiIconButton-sizeMedium css-10rq2ks-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg>
          trash.svg
        </svg>
      </button>
    </div>
  </div>
  <p
    className="MuiTypography-root MuiTypography-body2 css-15tpwza-MuiTypography-root"
  >
    Project description
  </p>
</a>
`;
